flink/src/main/java/org/apache/iceberg/flink/sink/FlinkSink.java

```diff
@@ -144,13 +144,14 @@ private Builder forRowData(DataStream<RowData> newRowDataInput) {
                                             MapFunction<T, RowData> mapper,
                                             TypeInformation<RowData> outputType) {
       this.inputCreator = newUidPrefix -> {
+        // Input stream order is crucial for some situation(e.g. in cdc case). Therefore, we need to set the parallelism
+        // of map operator same as it's input to keep map operator chaining it's input, and avoid rebalanced by default.
+        SingleOutputStreamOperator<RowData> inputStream = input.map(mapper, outputType)
+            .setParallelism(input.getParallelism());
```

Comment:
Sounds good to me ,  do we have any unit tests to address the regression issue ?

Reply:
I will add am unit test to cover this.

Reply:
I think we set different job parallelism in 'TestFlinkIcebergSinkV2' can already cover this, right?

https://github.com/apache/iceberg/pull/2898/files#diff-13e2e5b52d0effe51e1b470df77cb08b5ec8cc4f3a7f0fd4e51ee212fc83f76aL130